### ISSUE TITLE
Issue inspecting calling parent method

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#1255]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 3.0.12.1 |
| Language level | PHP 7.0                       |

I think this might be related to #1008.  Calls to parent class methods that don't use parent:: are being reported as needing to use $this->.

```
<?php

class ZA {

    protected function doSomething() {
    }

}


class ZB extends  ZA {

    protected function doSomething() {
        // $this->doSomething should be used instead
        // Should be parent::doSomething should be used instead
        ZA::doSomething();
    }

}

class ZC extends ZB {

    protected function doSomething() {
        // $this->doSomething should be used instead
        // Maybe should be some kind of warning about skipping parents
        ZA::doSomething();
    }

}
```

**Comment by orklah** [comment]
I had the same issue with my project. However, the code who do this seems suspicious to me, I'm not sure it's a good practice to call a dynamic method statically, even more when you skip the parent.

I was kind of surprised PHP accepts this so I didn't report it as a bug.

**Comment by kalessil** [comment]
I guess the problem is that QF turns methods into infinity loops. I'll re-check and if so, will release a patch shortly.

**Comment by kalessil** [comment]
Fixed! Patch-release is out.

**Comment by iakoder** [comment]
@kalessil Not sure if you care but this still gets caught out by mismatched case such as if ZB's method was named DoSomething and was trying to call ZA::doSomething().

**Comment by kalessil** [comment]
@iakoder : solved as well. Will be in the next regular release.
